pandas/core/groupby/generic.py

```diff
@@ -1740,18 +1746,18 @@ def count(self):
         ids, _, ngroups = self.grouper.group_info
         mask = ids != -1
 
-        val = (
+        vals = (
             (mask & ~_isna_ndarraylike(np.atleast_2d(blk.get_values())))
             for blk in data.blocks
         )
-        loc = (blk.mgr_locs for blk in data.blocks)
+        locs = (blk.mgr_locs for blk in data.blocks)
 
         counted = [
-            lib.count_level_2d(x, labels=ids, max_bin=ngroups, axis=1) for x in val
+            lib.count_level_2d(x, labels=ids, max_bin=ngroups, axis=1) for x in vals
         ]
-        blk = map(make_block, counted, loc)
+        blocks = [make_block(val, placement=loc) for val, loc in zip(counted, locs)]
```

Comment:
Just some minor cleanups above: pluralizing names + use list comprehension instead of map.

Reply:
i tend to prefer these too

Reply:
Can this just be a generator expression or does that fail? I see this is equivalent to existing code but maybe adds unnecessary overhead

Reply:
It would work, but wouldn't make any difference because it's passed into the BlockManager where it's stored.